### PR TITLE
Refactor the Reflog entries datetime comparisons

### PR DESCRIPTION
**Pull request by nulltoken**
We actually rely on `Assert.InRange` for this and usually ensure the entry has been created within the range of `[DateTimeOffset.Now - 5 seconds, DateTimeOffset.Now]`.

A better alternative (courtesy of @jamill), would be to retrieve the DateTimeOffset.Now before the action generating the reflog entry and ensuring the logged datetime uses this as the lower bound (ie.  `recordedOffset, DateTimeOffset.Now]`).

### CODE REVIEW

**Comment by nulltoken** [comment]
This little bugger failed one too many builds. Throwing some code at ir=t.

**Comment by nulltoken** [comment]
@dahlbyk @jamill Ready for review

File: LibGit2Sharp.Tests/TestHelpers/DateTimeOffsetExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace LibGit2Sharp.Tests.TestHelpers
+{
+    public static class DateTimeOffsetExtensions
+    {
+        public static DateTimeOffset TruncateMilliseconds(this DateTimeOffset dto)
+        {
+            // From http://stackoverflow.com/a/1005222/335418
```
Comment (nulltoken) [inline LibGit2Sharp.Tests/TestHelpers/DateTimeOffsetExtensions.cs]: /cc @martinwoodward

File: LibGit2Sharp.Tests/LibGit2Sharp.Tests.csproj [hunk 2]
```diff
@@ -136,7 +137,9 @@
   </ItemGroup>
   <ItemGroup>
     <None Include="app.config" />
-    <None Include="packages.config" />
+    <None Include="packages.config">
+      <SubType>Designer</SubType>
```
Comment (dahlbyk) [inline LibGit2Sharp.Tests/LibGit2Sharp.Tests.csproj]: SO MUCH :rage: AT THESE

**Comment by dahlbyk** [comment]
:+1:

File: LibGit2Sharp.Tests/LibGit2Sharp.Tests.csproj (hunk 2 above)
Comment (nulltoken) [inline LibGit2Sharp.Tests/LibGit2Sharp.Tests.csproj]: Good catch! Sorry I overlooked this. :blush:

**Comment by dahlbyk** [comment]
I almost wonder if we should capture `before` in a `BaseFixture` field. Not sure the slightly better accuracy of inline initialization is worth the extra noise in the tests.

**Comment by nulltoken** [comment]
@dahlbyk I thought about that, but there are tests that actually perform multiple operations, each of them generating a reflog entry, with a potential different lower boundary. For this kind of cases, we'd need some kind of call to reinitialize what `before` would contain.

We may end up with some tests having no time related variables at all, magically asserting durations and some others with a strange call to `ReinitHiddenTimer()` that might be difficult to understand to the reader.

I agree that this adds a ton of noise, but my main goal for this PR was to avoid restarting failed builds because of the way it was previously implemented.

Long story short, I'm very far from being against this proposal, but I'm miserably falling short at imagining a code piece that would make sense. Any ideas?

/cc @jamill

**Comment by dahlbyk** [comment]
A small objection, let's :shipit: and revisit if we find the noise unbearable?

**Comment by nulltoken** [comment]
:shipit:'d!

File: LibGit2Sharp.Tests/TestHelpers/DateTimeOffsetExtensions.cs (hunk 1 above)
Comment (nulltoken) [inline LibGit2Sharp.Tests/TestHelpers/DateTimeOffsetExtensions.cs]: /cc @ethomson
